app/controllers/profiles_controller.rb

```diff
@@ -14,51 +14,15 @@ class ProfilesController < ApplicationController
 
   def index
     if params[:search]
-      @profiles = profiles_for_search
-      # sum of search results concerning certain attributes
-      @aggs = @profiles.response.aggregations
-      @aggs_languages = @aggs[:lang][:buckets]
-      @aggs_cities = @aggs[:city][:buckets]
-      @aggs_states = @aggs[:state][:buckets]
-      @aggs_countries = @aggs[:country][:buckets]
-    elsif params[:tag_filter]&.present?
-      @tags = params[:tag_filter].split(/\s*,\s*/)
-      @profiles = profiles_with_tags
-      # redirect_to profiles_path(:anchor => "speakers")
-    elsif params[:category_id]
-      @profiles = profiles_for_category
+      search_with_search_params
     else
-      @profiles = profiles_for_index
-    end
-
-    @profiles_count = @profiles.total_count
-
-    # for the tags filter module that is available all the time at the profile index view
-    # is needed for the colors of the tags
-    @category =
-      if params[:category_id]
-        Category.find(params[:category_id])
-      elsif params[:tag_filter]
-        if params[:tag_filter].empty?
-          redirect_to profiles_url(anchor: "top"), notice: I18n.t('flash.profiles.no_tags_selected')
-          return
-        end
-        last_tag = params[:tag_filter].split(/\s*,\s*/).last
-        last_tag_id = ActsAsTaggableOn::Tag.where(name: last_tag).last.id
-        Category.select{|cat| cat.tag_ids.include?(last_tag_id)}.last
-      else
-        Category.first
+      if params[:tag_filter]&.empty? 
+        redirect_to profiles_url(anchor: "top"), notice: I18n.t('flash.profiles.no_tags_selected')
+        return 
```

Comment:
If I am reading this correctly, then before this change, there would be a page loading when a `category_id` query parameter would be set. With this change, if there is a `category_id` parameter, but no `tag_filter` parameter (for example, a page like https://speakerinnen.org/en/categories/7), then it'll end up redirecting to https://speakerinnen.org/en/profiles#top.
Am I correct or am I missing something? Also, if I'm correct, some test should fail.